lib/service/servicecfg/windows.go

```diff
@@ -50,24 +50,56 @@ type WindowsDesktopConfig struct {
 	// Discovery configures automatic desktop discovery via LDAP.
 	Discovery LDAPDiscoveryConfig
 
-	// Hosts is an optional list of static Windows hosts to expose through this
-	// service.
-	// Hosts is an optional list of static, AD-connected Windows hosts. This gives users
+	// ADHosts is an optional list of static, AD-connected Windows hosts. This gives users
 	// a way to specify AD-connected hosts that won't be found by the filters
 	// specified in Discovery (or if Discovery is omitted).
-	Hosts []utils.NetAddr
+	ADHosts []utils.NetAddr
 
 	// NonADHosts is an optional list of static Windows hosts to expose through this
 	// service. These hosts are not part of Active Directory.
 	NonADHosts []utils.NetAddr
 
+	// StaticHosts is an optional list of static Windows hosts to expose through this
+	// service.
+	StaticHosts []WindowsHost
+
 	// ConnLimiter limits the connection and request rates.
 	ConnLimiter limiter.Config
 	// HostLabels specifies rules that are used to apply labels to Windows hosts.
 	HostLabels HostLabelRules
 	Labels     map[string]string
 }
 
+func (w *WindowsDesktopConfig) AllStaticHosts() []WindowsHost {
+	var hosts []WindowsHost
+	hosts = append(hosts, w.StaticHosts...)
+	for _, host := range w.ADHosts {
+		hosts = append(hosts, WindowsHost{
+			Address: host,
+			AD:      true,
+		})
+	}
+	for _, host := range w.NonADHosts {
+		hosts = append(hosts, WindowsHost{
+			Address: host,
+			AD:      false,
+		})
+	}
+	return hosts
+}
```

Comment:
I think we should get rid of the `ADHosts` and `NonADHosts` fields on `WindowsDesktopConfig`, and instead do the conversion we're currently doing in `AllStaticHosts` when we copy the settings from the `FileConfig` here:

https://github.com/gravitational/teleport/blob/05dd39604225fb6978805dca29d7781baf9b5e5e/lib/config/configuration.go#L1969-L1983

That way we only keep these deprecated fields where they're absolutely necessary in the file config.

Reply:
That's a good suggestion, I like it.

Reply:
Yep, me too, I've refactored it